dali/kernels/imgproc/resample/resampling_impl.cuh

```diff
@@ -101,53 +107,176 @@ __device__ void ResampleHorz_Channels(
+ * but requires extra registers for the intermediate sums.
+ */
+template <int static_channels = -1, typename Dst, typename Src>
+__device__ void ResampleHorz_Channels(
+    ivec3 lo, ivec3 hi,
+    float src_x0, float scale,
+    Dst *__restrict__ out, ptrdiff_vec<2> out_strides,
+    const Src *__restrict__ in, ptrdiff_vec<2> in_strides, ivec3 in_shape, int dynamic_channels,
+    ResamplingFilter filter, int support) {
+  using resample_shared::coeffs;
+
+  int out_stride_y = out_strides.x;  // coordinates are shifted, because
+  int out_stride_z = out_strides.y;  // X stride is implicitly equal to the number of channels
+  int in_stride_y = in_strides.x;
+  int in_stride_z = in_strides.y;
+  int in_w = in_shape.x;
+
+  src_x0 += 0.5f * scale - 0.5f - filter.anchor;
+
+  const float filter_step = filter.scale;
+
+
+  const bool huge_kernel = support > 256;
+  const int coeff_base = huge_kernel ? 0 : threadIdx.x;
+  const int coeff_stride = huge_kernel ? 1 : blockDim.x;
+  const int channels = static_channels < 0 ? dynamic_channels : static_channels;
+
+  for (int j = lo.x; j < hi.x; j += blockDim.x) {
+    int dx = j + threadIdx.x;
+    const float sx0f = dx * scale + src_x0;
+    const int sx0 = huge_kernel ? __float2int_rn(sx0f) : __float2int_ru(sx0f);
+    float f = (sx0 - sx0f) * filter_step;
+    __syncthreads();
+    if (huge_kernel) {
+      for (int k = threadIdx.x + blockDim.x*threadIdx.y; k < support; k += blockDim.x*blockDim.y) {
+        float flt = filter(f + k*filter_step);
+        coeffs[k] = flt;
+      }
+    } else {
+      for (int k = threadIdx.y; k < support; k += blockDim.y) {
+        float flt = filter(f + k*filter_step);
+        coeffs[coeff_base + coeff_stride*k] = flt;
+      }
+    }
+    __syncthreads();
+
+    if (dx >= hi.x)
+      continue;
+
+    float norm = 0;
+    for (int k = 0; k < support; k++) {
+      norm += coeffs[coeff_base + coeff_stride * k];
+    }
+    norm = 1.0f / norm;
+
+    for (int z = threadIdx.z + lo.z; z < hi.z; z += blockDim.z) {
+      const Src *in_slice = &in[z * in_stride_z];
+      Dst *out_slice = &out[z * out_stride_z];
+
+      for (int i = threadIdx.y + lo.y; i < hi.y; i += blockDim.y) {
+        const Src *in_row = &in_slice[i * in_stride_y];
+        Dst *out_row = &out_slice[i * out_stride_y];
+
+        if (static_channels < 0) {
+          for (int c = 0; c < channels; c++) {
+            float tmp = 0;
+
+            for (int k = 0, coeff_idx=coeff_base; k < support; k++, coeff_idx += coeff_stride) {
+              int x = sx0 + k;
+              int xsample = x < 0 ? 0 : x >= in_w-1 ? in_w-1 : x;
+              float flt = coeffs[coeff_idx];
+              Src px = __ldg(in_row + channels * xsample + c);
+              tmp = fmaf(px, flt, tmp);
+            }
+
+            out_row[channels * dx + c] = ConvertSat<Dst>(tmp * norm);
+          }
+
+        } else {
+          float tmp[static_channels < 0 ? 1 : static_channels];  // NOLINT - not a variable length array
```

Comment:
static_channels is already >= 0 here.

Reply:
Tell that to compiler.

Reply:
I mean why not:
```
float tmp[static_channels];
```
but I probably missing something.